specification.yml

```diff
@@ -0,0 +1,174 @@
+asyncapi: 3.0.0
+info:
+  title: dp-search-data-importer
+  version: 1.1.0
+  description: |-
+    The *dp-search-data-importer* processes search-data-import events and then 
+    upserts the ElasticSearch index with them.
+  license:
+    name: MIT
+    url: https://opensource.org/license/MIT
+defaultContentType: avro/binary
+channels:
+  search-data-import:
+    messages:
+      search-data-import:
+        $ref: '#/components/messages/search-data-import'
+    description: This topic contains a message of processed search information, ready for import.
+operations:
+  search-data-import:
+    action: receive
+    channel:
+      $ref: '#/channels/search-data-import'
+    summary: Receive data for import to ElasticSearch
+    messages:
+      - $ref: '#/channels/search-data-import/messages/search-data-import'
+components:
+  messages:
+    search-data-import:
+      name: search-data-import
+      title: Item has been processed ready for search import
+      payload:
+        type: object
+        required:
+        - uid
+        - uri
+        - title
+        properties:
+          uid:
+            type: string
+            description: Unique ID by which to index in search
+          uri:
+            type: string
+            description: URI of upstream content item
+          data_type:
+            type: string
+            description: Content type ID, used for aggregations
+            enum:
+            - api_dataset_landing_page
+            - article
+            - article_download
+            - bulletin
+            - compendium_chapter
+            - compendium_data
+            - compendium_landing_page
+            - dataset
+            - dataset_landing_page
+            - home_page
+            - home_page_census
+            - product_page
+            - reference_tables
+            - release
+            - static_adhoc
+            - static_article
+            - static_foi
+            - static_landing_page
+            - static_methodology
+            - static_methodology_download
+            - static_page
+            - static_qmi
+            - taxonomy_landing_page
+            - timeseries
+            - visualisation
```

Comment:
Just wondering if this is too restrictive if we are opening up the service to new services. 🤔

Reply:
For the API, it doesn't necessarily matter as they would be surfaced and filtered on etc. For the controller however, it's much more restrictive and so we'd have to know what the content types are and map them in. Which is why I wanted to detail them.